autoscaler/controllers/gateway/config/coralogix.go

```diff
@@ -0,0 +1,59 @@
+package config
+
+import (
+	odigosv1 "github.com/keyval-dev/odigos/api/odigos/v1alpha1"
+	commonconf "github.com/keyval-dev/odigos/autoscaler/controllers/common"
+	"github.com/keyval-dev/odigos/common"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	coralogixPrivateKey = "CORALOGIX_PRIVATE_KEY"
+)
+
+type Coralogix struct{}
+
+func (c *Coralogix) DestType() common.DestinationType {
+	return common.CoralogixDestinationType
+}
+
+func (c *Coralogix) ModifyConfig(dest *odigosv1.Destination, currentConfig *commonconf.Config) {
+	if isTracingEnabled(dest) || isLoggingEnabled(dest) || isMetricsEnabled(dest) {
+		privateKey, exists := dest.Spec.Data[coralogixPrivateKey]
```

Comment:
Is this correct? I think secrets should be consumed by `"${CORALOGIX_PRIVATE_KEY}" and regular fields should be consumed by `dest.Spec.Data["CORALOGIX_DOMAIN"]`.
The reason is that Odigos automatically injects secret fields as env variables not to expose them in the regular Destination object.

Reply:
resolved this in later commits, thanks @edeNFed